lib/config.go

```diff
@@ -36,10 +50,24 @@ func DefaultConfig() *Config {
 			},
 		},
 		ResourceHandlers: []*ResourceHandlerConfig{},
+		Badger:           NewBadgerConfig(),
 		// DestPath: "sitemap.json",
 	}
 }
 
+// ErrNoBadgerConfig is the result of attempting to connect to a badgerDB
+// without one configured
+var ErrNoBadgerConfig = fmt.Errorf("badger is not configured")
+
+// BadgerDB returns the badger DB connection, creating a default-configured
+// badger store if one doesn't exist
+func (cfg *Config) BadgerDB() (*badger.DB, error) {
+	if cfg.Badger == nil {
+		return nil, ErrNoBadgerConfig
```

Comment:
In a previous version of this code, we *created* a new Badger config here. I think this change is mostly reasonable except it exposes an issue I hadn’t noticed before: `JSONConfigFromFilepath` completely overwrites instead of adding to the existing config. https://github.com/qri-io/walk/blob/e13838a20af9a993969e689c7d581846df265f18/lib/config.go#L43-L59

Because of that, it took me a bit of hunting to realize I now need to add  a `Badger` section to my config file… and that that section needs more than just the `Dir` and `ValueDir` properties. It doesn’t seem to match up with what the [Badger docs say is required](https://godoc.org/github.com/dgraph-io/badger#Options), so a after a couple false starts, I just copied the entirety of `badger.DefaultOptions` in there:

```js
{
    "Coordinator": /* ... */,
    "Queue": /* ... */,
    "RequestStore": /* ... */,
    "Workers": /* ... */,
    "ResourceHandlers": /* ... */,
    "Badger": {
      "Dir": "badger",
      "ValueDir": "badger",
      "ValueLogFileSize": 1073741823,
      "DoNotCompact":        false,
      "LevelOneSize":        268435456,
      "LevelSizeMultiplier": 10,
      "TableLoadingMode":    1,  // options.LoadToRAM, not so great to reference an enum/constant this way
      "ValueLogLoadingMode": 2,  // options.MemoryMap, not so great to reference an enum/constant this way

      "MaxLevels":               7,
      "MaxTableSize":            67108864,
      "NumCompactors":           3,
      "NumLevelZeroTables":      5,
      "NumLevelZeroTablesStall": 10,
      "NumMemtables":            5,
      "SyncWrites":              true,
      "NumVersionsToKeep":       1,

      "ValueLogMaxEntries": 1000000,
      "ValueThreshold":     32,
      "Truncate":           false
    }
}
```

Anyway, all that seems pretty painful. (Also: was there an easier way for me to have done this?) I think it’s probably `JSONConfigFromFilepath` that should change and the code you have here now should stay, but not really certain about that.

Barring any other changes, it seems like we should at least document this in an example config file.